modules/search-pipeline-common/src/main/java/org/opensearch/search/pipeline/common/SearchPipelineCommonModulePlugin.java

```diff
@@ -0,0 +1,31 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.search.pipeline.common;
+
+import org.opensearch.plugins.Plugin;
+import org.opensearch.plugins.SearchPipelinesPlugin;
+import org.opensearch.search.pipeline.Processor;
+
+import java.util.Map;
+
+/**
+ * Plugin providing common search request/response processors for use in search pipelines.
+ */
+public class SearchPipelineCommonModulePlugin extends Plugin implements SearchPipelinesPlugin {
+
+    /**
+     * No constructor needed, but build complains if we don't have a constructor with JavaDoc.
+     */
+    public SearchPipelineCommonModulePlugin() {}
```

Comment:
Why not just `SearchPipelineModulePlugin`, this `common` part looks a bit redundant (to me)?

Reply:
The "common" part was a mirror of `ingest-common` and its `IngestCommonModulePlugin`. (A common strategy I used for naming was `s/Ingest/SearchPipeline/`.) 

This module is intended as a repository for a "common" toolbox of generally-useful processors. Separately, we may later add modules with more specific use-cases -- like `ingest-user-agent` or `ingest-geoip`.

In the absence of those specific use-cases, the word "common" is weird and redundant. 

What do you think? Go for the "anticipate the future" approach or the "[you ain't gonna need it](https://en.wikipedia.org/wiki/You_aren%27t_gonna_need_it)" approach? (Following the YAGNI principle, I should probably remove "Common" from the class name for now, and add it back if/when it's needed.)

Reply:
Got it, I think it makes sense now to keep `Common`, since the similar paths have been taken